### PR TITLE
Optimise `__iter__`

### PR DESCRIPTION
**Pull request by DavidBuchanan314**
This uses an fstring to do the padded bit conversion, which I *think* is faster because it's a single operation (untested), but the more important change is that it avoids the need to reverse and then un-reverse the bits.

There could be further special-case optimisations involving to_bytes/from_bytes when the lane sizes are multiples of 8 bits.

### CODE REVIEW

File: test_simd.py [hunk 1]
```diff
@@ -216,4 +216,7 @@ def test_iter() -> None:
     a = A(0x_05_04_03_02_01, s)
     assert list(a) == [1, 2, 3, 4, 5] # note the "reversed" order
 
+    a = A(0x_056_034_012, simd.S(len=3, bv=8, bp=4))
+    assert list(a) == [0x12, 0x34, 0x56]
+
```
Comment (DavidBuchanan314) [inline test_simd.py:221]: I added this test case to try to catch confusions between bi/bp/bv etc.

**Comment by denballakh** [comment]
I think we need a benchmark that will help us know which operations are faster than others.
Double-reversing was unnecessary, sure. But I'm not sure if f-string conversion if faster than `bin(x)[2:]`. Needs to be tested.
I will probably merge this later after some investigation.

**Comment by DavidBuchanan314** [comment]
If the fstring ends up being slower, something to consider to speed up bin() is to offset the lookups by 2, rather than doing the `[2:]` (which I *think* incurs a buffer copy, but I'm not sure)

**Comment by denballakh** [comment]
I observed no significant difference. I guess they call the same function under the hood.

**Pull request merged**